src/server/pkg/storage/fileset/index/util.go

```diff
@@ -53,17 +53,17 @@ func PointsTo(idx *Index) (ids []chunk.ID) {
 	return ids
 }
 
-func resolveDataOps(idx *Index) {
-	if idx.FileOp.DataRefs == nil {
+func resolveParts(idx *Index) {
```

Comment:
What are the performance implications of not doing this in place? I think it would be clearer if this returned a new value.

Reply:
I don't think the performance hit would be much of a concern. I agree with this sentiment in general, but I think I will just leave this as a TODO because this is going to get re-written / thrown away if we move to the index entry per part model.

Reply:
OK. TODO is fine.